libz-rs-sys/src/gz.rs

```diff
@@ -109,7 +111,9 @@ unsafe fn gzopen_help(path: *const c_char, fd: c_int, mode: *const c_char) -> gz
     let Some(state) = ALLOCATOR.allocate_zeroed_raw::<GzState>() else {
         return ptr::null_mut();
     };
-    let state = state.cast::<GzState>().as_mut();
+    // Safety: the allocate_zeroed_raw call above ensures that the allocated block
+    // has the right size and alignment to be used as a GzState.
+    let state = unsafe { state.cast::<GzState>().as_mut() };
```

Comment:
A crucial detail here is that it is valid for `GzState` to be all zero bytes. Creating a reference also means we assert that this value is correctly initialized, which the `calloc` here handles for us.